### PR TITLE
feat: implement optional sample field for DRO update

### PR DESCRIPTION
**Pull request by baturayo**
This PR implements a sampling feature for the Distribution Reference Object (DRO) generation. In some cases, we were hitting the memory issues if we try to generate a DRO from large tables. Thanks to sampling approach, it will be doable to create DRO with limited resources.

### CODE REVIEW

File: soda/core/tests/cli/test_cli_update_distribution_file.py [hunk 1]
```diff
@@ -12,20 +12,64 @@
     test_data_source != "postgres",
```
Comment (tituskx) [inline soda/core/tests/cli/test_cli_update_distribution_file.py:12]: I would change this include testing for the other supported data sources, at least these ones: https://github.com/sodadata/soda-core/pull/1847/files#diff-1efbd7cdb05c3bdfb308ad8ec6791d31fd6d4bffd662e858dfd726f23e27e7ccR104-R106

**Review by tituskx — COMMENTED** [review]
Looks good! Left one comment

File: soda/core/tests/cli/test_cli_update_distribution_file.py (hunk 1 above)
Comment (baturayo) [reply soda/core/tests/cli/test_cli_update_distribution_file.py:12]: This part will be resolved when you merge your [PR](https://github.com/sodadata/soda-core/pull/1847) to main. I'll rewrite this logic after merging your PR with mine.

**Review by bastienboutonnet — COMMENTED** [review]
Leaving just a comment at this point since @baturayo and I DMed that it would make more sense for [@tituskx 's PR](https://github.com/sodadata/soda-core/pull/1847) to be merged by core team first and @baturayo will resolve the conflicts as the other way around would be more effort.

But consider this a tentative approval and when @tituskx PR is merged + conflicts resolved we'll properly review it agian.

**Review by bastienboutonnet — APPROVED** [review]
As far as I'm concerned this is good to me. Since you pinged @tituskx on the team channel, I will not mark this as ready to review. If he's ok with the PR then:
- remove ready for IA team
- mark PR as ready to review so that Vijay can merge it

Here's a "shipit cat" (it's a cat most of the time) but I am at the mercy of the randomisation gods.

File: soda/core/tests/cli/test_cli_update_distribution_file.py [hunk 2]
```diff
@@ -47,10 +39,81 @@ def mock_file_system_and_run_cli(
...
+    dro_file = dro_config.format(table_name=table_name)
+    mock_file_system_and_run_cli(mock_file_system, data_source_fixture, dro_file)
+
+
+@pytest.mark.skipif(
+    test_data_source not in DATA_SOURCES_WITH_DISTRIBUTION_CHECK_SUPPORT,
+    reason="Support for other data sources is experimental and not tested",
+)
+@pytest.mark.parametrize(
+    "distribution_type, expected_bins_and_weights",
```
Comment (tituskx) [inline soda/core/tests/cli/test_cli_update_distribution_file.py:114]: 🔥🔥🔥🔥🔥

File: soda/core/soda/cli/cli.py [hunk 3]
```diff
@@ -319,7 +319,13 @@ def update_dro(
     if filter is not None:
         filter_clause = f"WHERE {filter}"
 
+    sample = distribution_dict.get("sample")
+    sample_clause = sample if sample is not None else ""
+
     if dataset_name and column_name and distribution_type:
+        query = f"SELECT {column_name} FROM {dataset_name} {sample_clause} {filter_clause}"
```
Comment (tituskx) [inline soda/core/soda/cli/cli.py:326]: I guess the added whitespace between `dataset_name` and `filter_clause` doesn't mess up the sql right (if `sample_clause` is empty)?

**Review by tituskx — APPROVED** [review]
Thanks for adding the test coverage and refactoring them! I guess you tested the changes locally with CLI as well?

If so, then Leprechauns Guarding The Marshmallows!

**Pull request merged**